crates/core/app/src/app/mod.rs

```diff
@@ -151,6 +152,19 @@ impl App {
             }
         };
 
+        // Apply the state from `init_chain` and return the events generated during genesis. These
+        // cannot be emitted at this time due to a limitation of ABCI, but they will be emitted
+        // during the first BeginBlock.
+        let events = state_tx.apply().1;
+
+        // Commit the genesis events to the state.
+        let mut state_tx = self
+            .state
+            .try_begin_transaction()
+            .expect("state Arc should not be referenced elsewhere");
+        for (i, event) in events.into_iter().enumerate() {
+            state_tx.put_deferred_event(i, event);
+        }
```

Comment:
On reflection I have more concerns that this may be the wrong approach and I think we probably shouldn't do it.

Reply:
Is there another way we should handle it?

Reply:
@zbuc sorry, i got distracted doing the full writeup and was delayed. filled in the comment now

Reply:
So the idea is that we remove this specially cased event emission and then let the indexers use genesis JSON to bootstrap initial state? Sounds fine to me.

Reply:
@hdevalence indexers would also need to handle the genesis data checkpointed during upgrades.